### PR TITLE
[feat] llgo/c/hyper-related c lib

### PR DESCRIPTION
**Pull request by spongehah**
add new c libs:
`netdb`
`select`
`fddef`

### CODE REVIEW

File: c/netdb/netdb.go [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+	AiSockType  c.Int
+	AiProtocol  c.Int
+	AiAddrLen   c.Uint
+	AiCanOnName *c.Char
+	AiAddr      *socket.SockAddr
+	AiNext      *AddrInfo
+}
+
+//go:linkname GetAddrInfo C.getaddrinfo
+func GetAddrInfo(host *c.Char, port *c.Char, addrInfo *AddrInfo, result **AddrInfo) c.Int
```
Comment (luoliwoshang) [inline c/netdb/netdb.go]: The style of these functions needs to be changed. `GetAddrInfo` should be changed to `Getaddrinfo` to maintain consistency with the auto-generation tool, which does not expect to process word breaks in the original symbol (e.g., getaddrinfo) into camel case naming.

**Review by luoliwoshang — CHANGES_REQUESTED** [review]

**Comment by xushiwei** [comment]
> add new c libs: `netdb` `select` `fddef`

- merge package `fddef`, `select` into `c/syscall`;
- keep `FD_ISSET`, `FD_SET`, etc. (that is, change `FdIsset ` => `FD_ISSET`, `Fdset ` => `FD_SET`, etc.

**Comment by xushiwei** [comment]
> add new c libs: `netdb` `select` `fddef`

rename `c/socket` => `c/net` and merge package `netdb` into `c/net`.

**Comment by spongehah** [comment]
> > add new c libs: `netdb` `select` `fddef`
> 
> * merge package `fddef`, `select` into `c/syscall` (put code in `c/syscall/select.go`);
> * keep `FD_ISSET`, `FD_SET`, etc. (that is, change `FdIsset ` => `FD_ISSET`, `Fdset ` => `FD_SET`, etc.

! [image] (https://github.com/user-attachments/assets/52b8d38c-8cf3-4072-a664-9a101b7a10cc)
Here LLGoPackage is conflicting

**Comment by xushiwei** [comment]
> > > add new c libs: `netdb` `select` `fddef`
> > 
> > 
> > 
> > * merge package `fddef`, `select` into `c/syscall` (put code in `c/syscall/select.go`);
> > * keep `FD_ISSET`, `FD_SET`, etc. (that is, change `FdIsset ` => `FD_ISSET`, `Fdset ` => `FD_SET`, etc.
> 
> ! [image] (https://github.com/user-attachments/assets/52b8d38c-8cf3-4072-a664-9a101b7a10cc) Here LLGoPackage is conflicting

Merge package `select`, `fddef` to `c/sys`.

And

```go
const LLGoPackage = "link: c"
```

can be:

```go
const LLGoPackage = "link"
```

File: c/sys/select.go [hunk 2]
```diff
@@ -0,0 +1,31 @@
+package sys
+
+import (
+	"github.com/goplus/llgo/c/syscall"
```
Comment (xushiwei) [inline c/sys/select.go:4]: put `llgo/c/xxx` packages together

File: c/sys/select.go [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+	"github.com/goplus/llgo/c"
+)
+
+const (
+	LLGoFiles   = "_wrap/fddef.c"
+	LLGoPackage = "link"
+)
+
+// (TODO) merge to timeval
+type TimeVal struct {
```
Comment (xushiwei) [inline c/sys/select.go:16]: put `TimeVal` to package `llgo/c/time`

File: c/sys/select.go [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+	TvSec  c.Long
```
Comment (xushiwei) [inline c/sys/select.go:17]: remove `Tv` prefix

File: c/net/net.go [hunk 5]
```diff
@@ -110,6 +109,17 @@ type Hostent struct {
 	AddrList **c.Char // null-terminated array of addresses for the host
 }
 
+type AddrInfo struct {
+	AiFlags     c.Int
```
Comment (xushiwei) [inline c/net/net.go:113]: remove `Ai` prefix

**Pull request merged**